### PR TITLE
[RW-9148][risk=low] Filter by Disk Creator for list disk api 

### PR DESCRIPTION
**Pull request by yonghaoy**
Description:
`listDisksByProject` and `listAppsByProject` also returns apps/disks created by other people in the same workspace (if the caller can view those disks/apps). 

 similar to the old DiskApi: https://github.com/all-of-us/workbench/blob/main/api/src/main/java/org/pmiops/workbench/api/DiskController.java#L46)

That should work for both GCE and App PD: https://github.com/all-of-us/workbench/blob/main/api/src/main/java/org/pmiops/workbench/db/model/DbUser.java#L459-L477

But eventually we will want to support adding filter in listDisk request body after https://broadworkbench.atlassian.net/browse/IA-3823 done  

```
 curl -X 'GET' \                                                                                                                3 err | 16:31:37 
  'http://localhost:8081/v1/workspaces/aou-rw-local1-0ac92612/disks' \
  -H 'accept: application/json' -H "Authorization: Bearer $(gcloud auth print-access-token)"
[{"name":"all-of-us-pd-1-f4d5","status":"Ready","size":120,"diskType":"pd-standard","blockSize":4096,"appType":null,"isGceRuntime":true,"creator":"yonghao3@fake-research-aou.org","createdDate":"2022-12-14T20:52:52.646085Z","dateAccessed":"2022-12-14T21:01:17.300913Z"}]%   
```

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have added explanatory comments where the logic is not obvious
- [ ] I have run and tested this change locally, and my testing process is described here
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later
- [ ] If this includes an API change, I have run the E2E tests on this change against my local server with [yarn test-local](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples) because this PR won't be covered by the CircleCI tests 
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P)
- [ ] If this change impacts deployment safety (e.g. removing/altering APIs which are in use) I have documented these in the description
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and updated the appropriate API consumers
  * AoU UI
  * [Perf tests](https://github.com/broadinstitute/mcnulty/blob/develop/src/test/scala/services/AoU.scala)
  * [Researcher Directory export](https://github.com/all-of-us/workbench/wiki/Researcher-Directory-(RDR-export))
  * Cron tasks - for Offline*Controllers
  * SumoLogic - for EgressAlert

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/api/DisksController.java [hunk 1]
```diff
@@ -108,9 +113,14 @@ private List<Disk> findTheMostRecentActiveDisks(List<Disk> disksToValidate) {
...
     // incorrect state to help future debugging.
     // Disk maybe in incorrect state if having additional active state disks.
+    String pdNamePrefix = userProvider.get().getUserPDNamePrefix();
+
     List<Disk> activeDisks =
         disksToValidate.stream()
-            .filter(d -> ACTIVE_DISK_STATUS.contains(d.getStatus()))
+            .filter(
+                d ->
+                    ACTIVE_DISK_STATUS.contains(d.getStatus())
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/api/DisksController.java:122]: this should be plural: `ACTIVE_DISK_STATUSES`

File: api/src/main/java/org/pmiops/workbench/api/DisksController.java [hunk 2]
```diff
@@ -108,9 +113,14 @@ private List<Disk> findTheMostRecentActiveDisks(List<Disk> disksToValidate) {
...
+                        && d.getName().startsWith(pdNamePrefix))
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/api/DisksController.java:123]: could we use `Disk.creator` here instead?

Comment (yonghaoy) [reply api/src/main/java/org/pmiops/workbench/api/DisksController.java:123]: ah good point! I did see creator is propagated correctly!

**Review by jmthibault79 — APPROVED** [review]
This looks correct to me.  **I will also be merging this** because @yonghaoy is out, to unblock other work.

**Pull request merged**